src/components/editor/ArticleListController.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import FalcorController from "lib/falcor/FalcorController";
+import List from "./List";
+
+var pageLength = 50;
+
+export default class ArticleListController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUpdateSlugValue = this.handleUpdateSlugValue.bind(this);
+    this.safeSetState({
+      slugSearchValue: ""
+    })
+  }
+  static getFalcorPath(params) {
+    return [['articlesByTime', 'paginated', {from: pageLength*(params.page-1), length: pageLength}, ['title', 'slug']], ['articlesByTime', 'paginated', 'total']];
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let pathStart = "/articles/page/" + this.props.params.page + "/article/slug/";
+    browserHistory.push(pathStart + this.state.slugSearchValue + "/");
+    this.safeSetState({
+      slugSearchValue: ""
+    });
+  }
+
+  handleUpdateSlugValue(event) {
+    this.safeSetState({
+      slugSearchValue: event.target.value
+    });
+  }
+
+  handlePageChange(increment, event) {
+    let path = this.props.location.pathname;
+    path = path.split('/');
+    path[3] = (parseInt(path[3])+increment).toString();
+    browserHistory.push(path.join('/'));
+  }
+
+  render() {
+      // For Ling when code reviewing. I was choosing between two style decisions here when making the loading look as I wanted it
```

Comment:
The most important thing is how easy it is to follow and read. 200 lines of clear code is better than 50 lines of compact code. Like it's just quite difficult right now to figure out what happens, because you're calculating maxPage and firstIndex even when there is no data at all! I have no idea what the different cases are currently, and what is returned in each one.